include/dlaf/interface/cholesky.h

```diff
@@ -0,0 +1,105 @@
+
+template <typename T>
+void pxpotrf(char uplo, T* a, int m, int n, int mb, int nb, int lld, const MPI_Comm& communicator,
+             int nprow, int npcol) {
+  pika::resume();
+
+  // TODO: Check uplo
+  auto dlaf_uplo = (uplo == 'U' or uplo == 'u') ? blas::Uplo::Upper : blas::Uplo::Lower;
+
+  dlaf::comm::Communicator world(communicator);
+  DLAF_MPI_CHECK_ERROR(MPI_Barrier(world));
+
+  dlaf::comm::CommunicatorGrid communicator_grid(world, nprow, npcol, dlaf::common::Ordering::RowMajor);
+
+  dlaf::GlobalElementSize matrix_size(m, n);
+  dlaf::TileElementSize block_size(mb, nb);
+
+  dlaf::comm::Index2D src_rank_index(0, 0);  // WARN: Is this always the case?
+
+  dlaf::matrix::Distribution distribution(matrix_size, block_size, communicator_grid.size(),
+                                          communicator_grid.rank(), src_rank_index);
+
+  dlaf::matrix::LayoutInfo layout = colMajorLayout(distribution, lld);
+
+  dlaf::matrix::Matrix<T, dlaf::Device::CPU> matrix_host(std::move(distribution), layout, a);
+
+  {
+    MatrixMirror<T> matrix(matrix_host);
+
+    dlaf::factorization::cholesky<dlaf::Backend::Default, dlaf::Device::Default, T>(communicator_grid,
+                                                                                    dlaf_uplo,
+                                                                                    matrix.get());
+  }  // Destroy mirror
+
+  matrix_host.waitLocalTiles();
+
+  pika::suspend();
+}
+
+template <typename T>
+void pxpotrf(char uplo, int n, T* a, int ia, int ja, int* desca, int& info) {
+  utils::dlaf_check(uplo, desca, info);
+  if (info == -1)
+    return;
+  info = -1;  // Reset info to bad state
+
+  pika::resume();
+
+  auto dlaf_uplo = (uplo == 'U' or uplo == 'u') ? blas::Uplo::Upper : blas::Uplo::Lower;
+
+  auto [distribution, layout, communicator_grid] = blacs::dlaf_setup_from_desc(desca);
+
+  dlaf::matrix::Matrix<T, dlaf::Device::CPU> matrix_host(std::move(distribution), layout, a);
+
+  {
+    MatrixMirror<T> matrix(matrix_host);
+
+    dlaf::factorization::cholesky<dlaf::Backend::Default, dlaf::Device::Default, T>(communicator_grid,
+                                                                                    dlaf_uplo,
+                                                                                    matrix.get());
+  }  // Destroy mirror
+
+  matrix_host.waitLocalTiles();
+
+  pika::suspend();
+
+  info = 0;
+}
+
+extern "C" void cholesky_d(char uplo, double* a, int m, int n, int mb, int nb, int lld,
+                           const MPI_Comm& communicator, int nprow, int npcol);
+
+extern "C" void cholesky_s(char uplo, float* a, int m, int n, int mb, int nb, int lld,
+                           const MPI_Comm& communicator, int nprow, int npcol);
+
+extern "C" void pdpotrf(char uplo, int n, double* a, int ia, int ja, int* desca, int& info);
+
+extern "C" void pspotrf(char uplo, int n, float* a, int ia, int ja, int* desca, int& info);
```

Comment:
Sounds good for Cholesky, but what about `dlaf_eigensolver_eigensolver_d`? I have a slight preference for the former: `dlaf_cholesky_d`, `dlaf_eigensolver_d`, ...

Reply:
> but what about `dlaf_eigensolver_eigensolver_d`

Yeah, that doesn't look great... Is it a bad idea to come up with some naming scheme close to `pdpotrf` for the first two variants or is that going to cause confusion?

Reply:
I think it might be worth differentiating the ScaLAPACK-like API (which needs ScaLAPACK as a dependency, so we might want to make it conditional)? For now I went with the following, but open to further discussion:
```
dlaf_pdpotrf(uplo, n, a, ia, ja, desc_a, info)
dlaf_cholesky_d(dlaf_context, uplo, a, dlaf_desc_a)
```